datastream-server/src/test/java/com/linkedin/datastream/server/TestAssignmentTaskMapLogger.java

```diff
@@ -0,0 +1,147 @@
+/**
+ *  Copyright 2023 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.mockito.Mockito;
+
+import org.slf4j.Logger;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableSet;
+
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.contains;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+
+/**
+ * Tests for {@link AssignmentTaskMapLogger}
+ */
+public class TestAssignmentTaskMapLogger {
+  private static final DatastreamTask MOCK_TASK = mock(DatastreamTask.class);
+  private static final DatastreamTask MOCK_TASK_2 = mock(DatastreamTask.class);
+  private static final DatastreamTask MOCK_LARGE_TASK = Mockito.spy(new DatastreamTaskImpl());
+  private static final double SIZE_LIMIT_BYTES = 1024 * 1024;
+
+  @Test
+  public void testTasksLessThanSizeLimit() {
+    final Logger log = mock(Logger.class);
+    Map<String, Set<DatastreamTask>> taskMap = new HashMap<>();
+    taskMap.put("host1", ImmutableSet.of(MOCK_TASK));
+    taskMap.put("host2", ImmutableSet.of(MOCK_TASK, MOCK_TASK_2));
+
+    AssignmentTaskMapLogger logger = new AssignmentTaskMapLogger(log, SIZE_LIMIT_BYTES);
+    long startTime = System.currentTimeMillis();
+    logger.logAssignment(taskMap);
+    long endTime = System.currentTimeMillis();
+    long duration = endTime - startTime;
+    System.out.println("Log assignment duration: " + duration + " ms");
+
+    verify(log, times(3)).info(eq("{}={}"), contains("Live task"), anyString());
+
+    verifyNoMoreInteractions(log);
+  }
+
+  @Test
+  public void testTaskWithSizeExactlySizeLimit() {
+    final Logger log = mock(Logger.class);
+    Mockito.doReturn(createCustomSizeString(SIZE_LIMIT_BYTES)).when(MOCK_LARGE_TASK).toString();
```

Comment:
how long does this test takes to pass ? maybe you can reduce the SIZE_LIMIT_BYTES to 1KB or lesser as well for tests and builds to pass quickly ?

Reply:
build took 12 seconds but i can make it smaller

Reply:
I decided to make the size limit 4KB because if the limit is too small the buffer adjustment makes it so that message is split up even more than necessary, which is expected in practice because splitting more is more safe, but the unit test expectation would be less intuitive. For example, if the size limit was 2KB and we have a message of size 4KB, instead of the message getting split into 3 parts (4KB should be split into is 2 log lines but we "round up" so we split more to be safe), the buffer adjustment of 1KB would make it so that only 1KB of the message would get logged at a time, and so it actually gets split into at least 4 log lines, which is fine in practice but might be harder to understand when reading the unit test.